lighthouse-core/test/gather/driver-test.js

```diff
@@ -305,6 +305,123 @@ describe('.evaluateAsync', () => {
   });
 });
 
+describe('.evaluate', () => {
+  it('transforms parameters into an expression given to Runtime.evaluate', async () => {
+    connectionStub.sendCommand = createMockSendCommandFn()
+      .mockResponse('Runtime.evaluate', {result: {value: 1}});
+
+    /** @param {number} value */
+    function main(value) {
+      return value;
+    }
+    const value = await driver.evaluate(main, {args: [1]});
+    expect(value).toEqual(1);
+
+    const {expression} = connectionStub.sendCommand.findInvocation('Runtime.evaluate');
+    const expected = `
+(function wrapInNativePromise() {
+        const __nativePromise = globalThis.__nativePromise || Promise;
+        const URL = globalThis.__nativeURL || globalThis.URL;
+        return new __nativePromise(function (resolve) {
+          return __nativePromise.resolve()
+            .then(_ => (() => {
+      
+      function main(value) {
+      return value;
+    }
+      return main(1);
+    })())
+            .catch(function wrapRuntimeEvalErrorInBrowser(err) {
+  err = err || new Error();
+  const fallbackMessage = typeof err === 'string' ? err : 'unknown error';
+  return {
+    __failedInBrowser: true,
+    name: err.name || 'Error',
+    message: err.message || fallbackMessage,
+    stack: err.stack || new Error().stack
+  };
+})
+            .then(resolve);
+        });
+      }())`.trim();
+    expect(expression).toBe(expected);
+    expect(await eval(expression)).toBe(1);
+  });
+
+  it('transforms parameters into an expression (basic)', async () => {
+    const mockFn = driver._evaluateInContext = jest.fn()
```

Comment:
not essential, but maybe add a comment that this is mocking this to skip the promise/error-catching wrapper?